src/main/dynamo/operations/transactWriteItems.ts

```diff
@@ -1,16 +1,18 @@
 import { DynamoDB } from "aws-sdk"
+import { v4 as generateUUID } from "uuid"
 import { PartitionAndSortKey } from "../keys"
 import { TaggedModel } from "../types"
 import { MaybeEncryptedItem } from "../util"
 import { BaseParams } from "./BaseParams"
 
 export interface TransactWriteItemParams<T extends TaggedModel> extends BaseParams {
+  clientRequestToken?: string
   putItems?: MaybeEncryptedItem<T>[]
   deleteItems?: PartitionAndSortKey<T>[]
 }
 
 export async function transactWriteItems<T extends TaggedModel>(params: TransactWriteItemParams<T>): Promise<void> {
-  const { table, client, putItems = [], deleteItems = [] } = params
+  const { table, client, clientRequestToken = generateUUID(), putItems = [], deleteItems = [] } = params
```

Comment:
A better implementation, I didn't have time for right now, would be to calculate a stable hash of the input records ad use that as the default `clientRequestToken`